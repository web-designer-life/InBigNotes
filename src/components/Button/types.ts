enum BUTTON_TYPES {
    BUTTON = 'button',
    SUBMIT = 'submit',
    RESET = 'reset',
};

export default BUTTON_TYPES;