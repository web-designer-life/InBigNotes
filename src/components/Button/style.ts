import styled, { css } from 'styled-components';

const StyledButton = styled.button`
	min-width: 270px;
    height: 80px;
    border-radius: 20px;
    color: #FFFFFF;
    padding: 0 75px 0 75px;
    transition-duration: .1s;
    font-family: 'Lora', 'Arial', sans-serif;
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    &:hover {
        background-color: #FFFFFF;
    }
    &:disabled {
        opacity: .5;
        border: 5px solid #808080;
        background-color: #808080;
        &:hover { 
            background-color: #808080;
            color: #FFFFFF;
            cursor: default;
        }
    }
    ${(props) => (props.color === 'green' && css`
        border: 5px solid #00AB00;
        background-color: #00AB00;
        &:hover {
            color: #00AB00;
        }
    `)}
    ${(props) => (props.color === 'red' && css`
        border: 5px solid #FF0000;
        background-color: #FF0000;
        &:hover {
            background-color: #FFFFFF;
            color: #FF0000;
        }
    `)}
`;

export default StyledButton;
