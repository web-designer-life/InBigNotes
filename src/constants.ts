export enum ROUTES {
	HOME = '/',
    NOTE = '/note',
	NOTE_CREATE = '/note/create',
};

export enum BUTTON_TYPES {
    BUTTON = 'button',
    SUBMIT = 'submit',
    RESET = 'reset',
};

export enum BUTTON_COLORS {
    GREEN = 'green',
    RED = 'red',
};

export enum BUTTON_TEXTS {
    CREATE = 'Create',
    DELETE = 'Delete',
    BACK = 'Back',
    SAVE = 'Save',
    UPDATE = 'Update',
    CANCEL = 'Cancel',
    CONFIRM = 'Confirm',
};

export enum MODAL_TEXTS {
    DELETE = 'Do you want to delete selected notes?',
    BACK = 'You have changes in your note. Do you want to exit to the main menu without saving?',
    SAVE = 'Do you want to save your note?',
    UPDATE = 'Do you want to update your note?',
    CANCEL = 'You have changes in your note. Do you want to cancel the changes?',
};

export const SOURCE = {
    ROUTES: {
        HOME: '/',
        NOTE: '/note',
        NOTE_CREATE: '/note/create',
    },
    BUTTONS: {
        TYPES: {
            BUTTON: 'button',
            SUBMIT: 'submit',
            RESET: 'reset',
        },
        COLORS: {
            GREEN: 'green',
            RED: 'red',
        },
        TEXTS: {
            CREATE: 'Create',
            DELETE: 'Delete',
            BACK: 'Back',
            SAVE: 'Save',
            UPDATE: 'Update',
            CANCEL: 'Cancel',
            CONFIRM: 'Confirm',
        },
    },
    MODAL: {
        TEXTS: {
            DELETE: 'Do you want to delete selected notes?',
            BACK: 'You have changes in your note. Do you want to exit to the main menu without saving?',
            SAVE: 'Do you want to save your note?',
            UPDATE: 'Do you want to update your note?',
            CANCEL: 'You have changes in your note. Do you want to cancel the changes?',
        },
    },
}
