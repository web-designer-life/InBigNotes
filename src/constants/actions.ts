const ACTIONS = {
    SAVE: 'Save',
    UPDATE: 'Update',
};

export default ACTIONS;
