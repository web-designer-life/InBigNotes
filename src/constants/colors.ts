const COLORS = {
    BUTTON: {
        GREEN: 'green',
        RED: 'red',
    },
    WHITE: '#FFFFFF',
    GRAY: '#808080',
    STEEL_GRAY: '#5A5A5A',
    BLACK: '#000000',
    RED: '#FF0000',
    GREEN: '#00AB00',
    BLUE: '#0047FF',
};

export default COLORS;
