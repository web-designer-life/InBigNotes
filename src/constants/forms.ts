export const FORMS = {
    ADD_OR_UPDATE_NOTE_FORM: 'addOrUpdateNote',
};
