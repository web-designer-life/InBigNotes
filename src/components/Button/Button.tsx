import React, { Component } from 'react';
import { BUTTON_TYPES } from '../../constants';
import StyledButton from './style';

interface Props {
    form?: string,
    type: BUTTON_TYPES,
    disabled?: boolean,
    onClick?: () => void,
    text: string,
    color: string,
};

export default class Button extends Component<Props> {
    constructor(props) {
        super(props);

        this.state = {};
    }

    render() {
        const {
            type,
            disabled,
            onClick,
            text,
            color,
        } = this.props;

        return (
            <StyledButton
                type={type}
                disabled={disabled}
                onClick={onClick}
                color={color}
            >
                {text}
            </StyledButton>
        )
    }
};
