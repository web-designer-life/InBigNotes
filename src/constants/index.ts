import { COLORS } from './colors';
import { FORMS } from './forms';
import { ROUTES } from './routes';
import { TEXTS } from './texts';
import { BUTTON_TYPES, ACTIONS } from './types';

export {
    COLORS,
    FORMS,
    ROUTES,
    TEXTS,
    BUTTON_TYPES, 
    ACTIONS,
}