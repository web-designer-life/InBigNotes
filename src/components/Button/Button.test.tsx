import React from 'react'
import Button from './Button';
import { shallow } from 'enzyme';
import { BUTTON_TYPES, COLORS, TEXTS } from '../../constants';

const greenButtonProps = {
    type: BUTTON_TYPES.BUTTON,
    disabled: false,
    onClick: () => {},
    text: TEXTS.BUTTON.HOME,
    color: COLORS.GREEN,
};

describe("Button component", () => {
    it("should render green Button component", () => {
        const component = shallow(<Button {...greenButtonProps} />);
        expect(component).toMatchSnapshot();
    });

    it("should call onClick method", () => {
        const mockCallBack = jest.fn();
        const component = shallow(
            <Button 
                onClick={mockCallBack} 
                type={BUTTON_TYPES.BUTTON}
                disabled={false}
                text={TEXTS.BUTTON.BACK}
                color={COLORS.RED}
            />
        );
        expect(mockCallBack.mock.calls.length).toBe(0);
        component.simulate("click");
        expect(mockCallBack.mock.calls.length).toBe(1);
    });
});
