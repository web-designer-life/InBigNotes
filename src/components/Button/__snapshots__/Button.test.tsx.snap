// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render green Button component 1`] = `
<styled.button
  color="#00AB00"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Go home
</styled.button>
`;
